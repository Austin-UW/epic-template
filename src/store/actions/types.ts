import { TVariant } from 'src/types/state'
import { TBaseProject } from 'src/types/project'
import { TTask } from 'src/types/task'

/** snackbar */
export type TOpenSnackbar = {
  type: 'OPEN_SNACKBAR'
  message: string
  variant: TVariant
}
export type TCloseSnackbar = { type: 'CLOSE_SNACKBAR' }

export type SnackbarAction = TOpenSnackbar | TCloseSnackbar | TCreateProject

/** loading */
export type TStartLoading = { type: 'START_LOADING' }
export type TStopLoading = { type: 'STOP_LOADING' }

export type LoadingAction = TStartLoading | TStopLoading
/** project */

export type TCreateProject = { type: 'CREATE_PROJECT'; project: TBaseProject }
export type TDeleteProject = { type: 'DELETE_PROJECT'; id: number }
export type TRenameProject = {
  type: 'RENAME_PROJECT'
  id: number
  name: string
}

export type TCreateColumn = {
  type: 'CREATE_COLUMN'
  isCompletedColumn: boolean
  name: string
  projectId: number
}

export type TMoveColumn = {
  type: 'MOVE_COLUMN'
  projectId: number
  columnId: number
  direction: 'Back' | 'Forward'
}

/** both project and column reducer use these */
type ProjectAndColumnAction = TCreateColumn | TMoveColumn | TCreateProject
export type ProjectAction =
  | TDeleteProject
  | TDeleteColumn
  | TRenameProject
  | ProjectAndColumnAction
/** column */
export type TMoveTaskVertical = {
  type: 'MOVE_TASK_VERTICALLY'
  id: string
  newIndex: number
  columnId: string
}

export type TChangeTasksColumn = {
  type: 'CHANGE_TASKS_COLUMN'
  id: string
  newColumnId?: number
  oldColumnId?: number
  newIndex: number
}

export type TDeleteColumn = { type: 'DELETE_COLUMN'; id: number }

export type ColumnAction =
  | ProjectAndColumnAction
  | TCreateColumn
  | TDeleteColumn
  | TChangeTasksColumn
  | TMoveTaskVertical
  | TaskAndColumn

type TaskAndColumn = TCreateTask
/** task */

export type TCompleteSubTask = {
  type: 'COMPLETE_SUBTASK'
  taskId: number
  subTaskId: number
}

export type TDeleteTask = { type: 'DELETE_TASK'; id: number }

export type TEditTask = {
  type: 'EDIT_TASK'
  id: number
  task: TTask
}

export type TCreateTask = {
  type: 'CREATE_TASK'
  name: string
  columnI: number
  color: string
  dueDate?: Date
}

export type TaskAction =
  | TDeleteTask
  | TEditTask
  | TaskAndPomodoro
  | TCompleteSubTask
  | TaskAndColumn
  | { type: 'CHECK_TASK_OVERDUE' }

type TaskAndPomodoro = TTick
/** pomodoro */
export type TStopTimer = { type: 'STOP_TIMER' }
export type TStartTimer = { type: 'START_TIMER' }
export type TChangeWorkTime = {
  type: 'CHANGE_WORK_TIME'
  operator: '+' | '-'
  minutes: number
}
export type TChangeBreakTime = {
  type: 'CHANGE_BREAK_TIME'
  operator: '+' | '-'
  minutes: number
}
export type TTick = {
  type: 'TICK'
  taskId?: number
}

export type TResetPomodoro = {
  type: 'RESET_POMODORO'
}

export type TIncrementDailyTime = { type: 'INCREMENT_DAILY_TIME' }

export type TSelectPomodoroTask = {
  type: 'SELECT_POMODORO_TASK'
  taskId: null | string
}

export type TToggleSelectingTask = {
  type: 'TOGGLE_SELECTING_TASK'
}

export type TTickStopwatch = {
  type: 'TICK_STOPWATCH'
}

export type TToggleStopwatch = {
  type: 'TOGGLE_STOPWATCH'
}

export type TResetStopwatch = {
  type: 'RESET_STOPWATCH'
}

export type TSetTimeDev = { type: 'SET_TIME_DEV' }

export type PomodoroAction =
  | TStopTimer
  | TStartTimer
  | TChangeBreakTime
  | TChangeWorkTime
  | TResetPomodoro
  | TaskAndPomodoro
  | TIncrementDailyTime
  | TSelectPomodoroTask
  | TToggleSelectingTask
  | TSetTimeDev
  | TTickStopwatch
  | TToggleStopwatch
  | TResetStopwatch

export type TAction = Readonly<
  | SnackbarAction
  | LoadingAction
  | ProjectAction
  | TaskAction
  | ColumnAction
  | PomodoroAction
>
