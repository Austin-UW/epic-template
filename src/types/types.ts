import { MapStateToProps } from 'react-redux'

import { TState } from './state'

export type TAuthType = 'Register' | 'Login'
export type MSProps<StateProps> = MapStateToProps<StateProps, {}, TState>
import { ChangeEvent } from 'react'

export type Change = ChangeEvent<HTMLInputElement>
