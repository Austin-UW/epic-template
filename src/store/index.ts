export { authUserU } from './thunks/auth'

export { startLoadingA, stopLoadingA } from './actions/loading'
export { openSnackbarA, closeSnackbarA } from './actions/snackbar'
export { createProjectA } from './actions/project'

export { reducer } from './reducers/reducer'
export { store } from './store'
export { defaultState } from './defaultState'
