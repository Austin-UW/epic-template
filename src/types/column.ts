export type TColumn = {
  readonly id: number
  name: string
  isCompletedColumn?: boolean
  taskIds: number[]
}
export type TColumns = { [id: string]: TColumn }
