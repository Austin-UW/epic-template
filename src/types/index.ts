export { Change } from './react'
export { TSnackbar, TVariant } from './snackbar'
export { TUser } from './user'
export {
  TAction,
  TOpenSnackbar,
  TCloseSnackbar,
  TSetCurrentUser,
  TStartLoading,
  TStopLoading,
  TToggleMenu,
  TReset
} from './action'
export { TAuthType, MSProps } from './types'
export { TState } from './state'
