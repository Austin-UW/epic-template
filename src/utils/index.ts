import { cloneDeep } from 'lodash'

export const formalize = (str: string): string => {
  return str
    .toLowerCase()
    .replace(/,/g, '')
    .replace(/ /g, '')
}

export const filterItemsFromIds = <T = any>(
  ids: number[],
  items: { [id: string]: any } & T // add index signature
): T => {
  const newItems = cloneDeep(items) // fuckin changed in past :()
  Object.keys(newItems)
    .map(key => parseInt(key, 10)) // Object.keys returns strings nomatter what
    .forEach((key: any) => {
      if (!ids.includes(key)) {
        delete newItems[key]
      }
    })
  return newItems
}

export const getHighestId = (obj: { [i: number]: any }): number => {
  const result = Math.max(...Object.keys(obj).map(key => parseInt(key, 10)))
  if (result === -Infinity) {
    return -1
  }
  return result
}

export const getHighestIdElement = (obj: { [i: number]: any }) => {
  const highestId = getHighestId(obj)
  const highestIdElement = obj[highestId] // return the value of the highest

  return highestIdElement
}

export const getDaysInMonth = (month: number, year: number): Date[] => {
  // Since no month has fewer than 28 days
  const date: Date = new Date(year, month, 1)
  const days: Date[] = []
  while (date.getMonth() === month) {
    days.push(new Date(date))
    date.setDate(date.getDate() + 1)
  }
  return days
}
