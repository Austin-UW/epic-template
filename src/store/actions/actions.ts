import { TAction, TUser } from 'src/types'

export const setCurrentUserA = (user: TUser): TAction => {
  return {
    type: 'SET_CURRENT_USER',
    user
  }
}
export const toggleA = (open: boolean): TAction => ({ type: 'TOGGLE_MENU', open })
