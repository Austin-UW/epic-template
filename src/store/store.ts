import { createStore, applyMiddleware, Store } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import thunk from 'redux-thunk'
import { reducer } from '.'
import { defaultState } from './defaultState'
import { TState } from '../types/state'
import { incrementDailyTimeA } from './actions/pomodoro'
import { TAction } from '@store/actions/types'

export const loadState = (): undefined | TState => {
  try {
    const serializedState = localStorage.getItem('newerState')
    if (serializedState === null) {
      return undefined
    }
    const state: TState = JSON.parse(serializedState)
    Object.keys(state.tasks).map(id => {
      const task = state.tasks[id]
      if (task.dueDate) {
        const date = new Date()
        date.setTime(Date.parse(state.tasks[id].dueDate as any))
        state.tasks[id].dueDate = date
      }
    })
    return state
  } catch (err) {
    return undefined
  }
}

const saveState = (state: TState) => {
  state = {
    ...state,
    pomodoro: { ...state.pomodoro, paused: true }
  }
  try {
    const serializedState = JSON.stringify(state)
    localStorage.setItem('newerState', serializedState)
  } catch (err) {}
}

export const store: Store<TState, TAction> = createStore(
  reducer,
  defaultState,
  composeWithDevTools(applyMiddleware(thunk))
)

store.subscribe(() => {
  const pastAction = store.getState().pastAction
  const pastState = store.getState()
  if (pastAction.type === 'TICK') {
    if (pastState.pomodoro.currSeconds % 60 === 0) {
      store.dispatch(incrementDailyTimeA())
      store.dispatch({ type: 'CHECK_TASK_OVERDUE' })
      saveState(pastState) // save pomodoro time left every minute
    }
  } else if (pastAction.type === 'TICK_STOPWATCH') {
    if (pastState.pomodoro.stopWatch.time % 60 === 0) {
      store.dispatch({ type: 'CHECK_TASK_OVERDUE' })
      saveState(pastState)
    }
  } else if (pastAction.type !== 'CHECK_TASK_OVERDUE') {
    // dont let itself call itself that not good
    // store.dispatch({ type: 'CHECK_TASK_OVERDUE' })
    saveState(pastState)
  }
})
