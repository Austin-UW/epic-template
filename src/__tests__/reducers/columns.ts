import { defaultState, store, createProjectA } from '../../store'
import { TColumns } from '../../types/column'
import { createColumnA, deleteColumnA } from '../../store/actions/column'
import { getHighestId } from '../../utils'
import {
  createTaskA,
  moveTaskVerticalA,
  changeTasksColumn
} from '../../store/actions/task'

const getState = (): TColumns => store.getState().columns

beforeEach(() => {
  store.dispatch({ type: '@@INIT' } as any)
})

describe('columns reducer', () => {
  test('@@INIT', () => {
    expect(getState()).toEqual(defaultState.columns)
    expect(getState()).toMatchSnapshot()
  })
  test('CREATE_COLUMN', () => {
    const PROJECT_ID = 1
    store.dispatch(createColumnA('number 1', PROJECT_ID))
    // go through all id keys, turn them into natural numbers, and get maximum
    const highestId: number = Math.max(
      ...Object.keys(getState()).map(id => parseInt(id, 10))
    )
    const created = getState()[highestId]
    expect(store.getState().projects[PROJECT_ID].columnIds).toContain(highestId)
    expect(created).toEqual({
      id: getHighestId(getState()),
      name: 'number 1',
      isCompletedColumn: false,
      taskIds: []
    })
    expect(getState()).toMatchSnapshot()
  })
  test('CREATE_PROJECT', () => {
    const projectsBefore = Object.values(store.getState().projects)
    const stateBefore = Object.values(getState())
    const COLUMNS_ADDED_DEFAULT = 3
    store.dispatch(createProjectA(Object.values(defaultState.projects)[0]))
    expect(Object.values(store.getState().projects).length).toBe(
      projectsBefore.length + 1
    )
    expect(Object.values(getState()).length).toBe(
      stateBefore.length + COLUMNS_ADDED_DEFAULT
    )
    expect(getState()).toMatchSnapshot()
  })
  test('DELETE_COLUMN', () => {
    expect(getState()[9]).toBeDefined()
    store.dispatch(deleteColumnA(9) as any)
    expect(getState()[9]).not.toBeDefined()
    expect(getState()).toMatchSnapshot()
  })
  test('CREATE_TASK', () => {
    const COLUMN_ID = 1
    const before = getState()
    store.dispatch(createTaskA('Task Name', COLUMN_ID, 'BLUE', new Date()))
    expect(getState()[COLUMN_ID].taskIds).toHaveLength(
      before[COLUMN_ID].taskIds.length + 1
    )
    expect(getState()).toMatchSnapshot()
  })
  test('CHANGE_TASKS_COLUMN', () => {
    const TASK_ID = 1
    const NEW_COL_ID = 2
    const OLD_COL_ID = 0
    const NEW_INDEX = 1

    expect(getState()[NEW_COL_ID].taskIds).not.toContain(TASK_ID)

    store.dispatch(
      changeTasksColumn(TASK_ID.toString(), NEW_COL_ID, OLD_COL_ID, NEW_INDEX)
    )
    expect(getState()[OLD_COL_ID].taskIds).not.toContain(TASK_ID)
    expect(getState()[NEW_COL_ID].taskIds).toContain(TASK_ID)

    expect(getState()).toMatchSnapshot()
  })
  test('MOVE_TASK_VERTICALLY', () => {
    const TASK_ID = 1
    const NEW_INDEX = 1
    const COLUMN_ID = '0'
    expect(getState()[COLUMN_ID].taskIds[NEW_INDEX]).not.toBe(TASK_ID)
    store.dispatch(moveTaskVerticalA(TASK_ID.toString(), NEW_INDEX, COLUMN_ID))
    expect(getState()[COLUMN_ID].taskIds[NEW_INDEX]).toBe(TASK_ID)

    expect(getState()).toMatchSnapshot()
  })
})
