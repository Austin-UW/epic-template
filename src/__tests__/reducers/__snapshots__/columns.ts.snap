// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`columns reducer @@INIT 1`] = `
Object {
  "0": Object {
    "id": 0,
    "name": "To-do",
    "taskIds": Array [
      0,
      1,
      2,
    ],
  },
  "1": Object {
    "id": 1,
    "name": "In Progress",
    "taskIds": Array [
      3,
      4,
    ],
  },
  "2": Object {
    "id": 2,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [
      5,
    ],
  },
  "3": Object {
    "id": 3,
    "name": "To-do",
    "taskIds": Array [
      6,
    ],
  },
  "4": Object {
    "id": 4,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "5": Object {
    "id": 5,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
  "6": Object {
    "id": 6,
    "name": "To-do",
    "taskIds": Array [],
  },
  "7": Object {
    "id": 7,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "8": Object {
    "id": 8,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
}
`;

exports[`columns reducer CHANGE_TASKS_COLUMN 1`] = `
Object {
  "0": Object {
    "id": 0,
    "name": "To-do",
    "taskIds": Array [
      0,
      2,
    ],
  },
  "1": Object {
    "id": 1,
    "name": "In Progress",
    "taskIds": Array [
      3,
      4,
      7,
    ],
  },
  "10": Object {
    "id": 10,
    "name": "To-do",
    "taskIds": Array [],
  },
  "11": Object {
    "id": 11,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "12": Object {
    "id": 12,
    "isCompletedColumn": true,
    "name": "Done",
    "taskIds": Array [],
  },
  "2": Object {
    "id": 2,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [
      5,
      1,
    ],
  },
  "3": Object {
    "id": 3,
    "name": "To-do",
    "taskIds": Array [
      6,
    ],
  },
  "4": Object {
    "id": 4,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "5": Object {
    "id": 5,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
  "6": Object {
    "id": 6,
    "name": "To-do",
    "taskIds": Array [],
  },
  "7": Object {
    "id": 7,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "8": Object {
    "id": 8,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
}
`;

exports[`columns reducer CREATE_COLUMN 1`] = `
Object {
  "0": Object {
    "id": 0,
    "name": "To-do",
    "taskIds": Array [
      0,
      1,
      2,
    ],
  },
  "1": Object {
    "id": 1,
    "name": "In Progress",
    "taskIds": Array [
      3,
      4,
    ],
  },
  "2": Object {
    "id": 2,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [
      5,
    ],
  },
  "3": Object {
    "id": 3,
    "name": "To-do",
    "taskIds": Array [
      6,
    ],
  },
  "4": Object {
    "id": 4,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "5": Object {
    "id": 5,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
  "6": Object {
    "id": 6,
    "name": "To-do",
    "taskIds": Array [],
  },
  "7": Object {
    "id": 7,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "8": Object {
    "id": 8,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
  "9": Object {
    "id": 9,
    "isCompletedColumn": false,
    "name": "number 1",
    "taskIds": Array [],
  },
}
`;

exports[`columns reducer CREATE_PROJECT 1`] = `
Object {
  "0": Object {
    "id": 0,
    "name": "To-do",
    "taskIds": Array [
      0,
      1,
      2,
    ],
  },
  "1": Object {
    "id": 1,
    "name": "In Progress",
    "taskIds": Array [
      3,
      4,
    ],
  },
  "10": Object {
    "id": 10,
    "name": "To-do",
    "taskIds": Array [],
  },
  "11": Object {
    "id": 11,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "12": Object {
    "id": 12,
    "isCompletedColumn": true,
    "name": "Done",
    "taskIds": Array [],
  },
  "2": Object {
    "id": 2,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [
      5,
    ],
  },
  "3": Object {
    "id": 3,
    "name": "To-do",
    "taskIds": Array [
      6,
    ],
  },
  "4": Object {
    "id": 4,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "5": Object {
    "id": 5,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
  "6": Object {
    "id": 6,
    "name": "To-do",
    "taskIds": Array [],
  },
  "7": Object {
    "id": 7,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "8": Object {
    "id": 8,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
  "9": Object {
    "id": 9,
    "isCompletedColumn": false,
    "name": "number 1",
    "taskIds": Array [],
  },
}
`;

exports[`columns reducer CREATE_TASK 1`] = `
Object {
  "0": Object {
    "id": 0,
    "name": "To-do",
    "taskIds": Array [
      0,
      1,
      2,
    ],
  },
  "1": Object {
    "id": 1,
    "name": "In Progress",
    "taskIds": Array [
      3,
      4,
      7,
    ],
  },
  "10": Object {
    "id": 10,
    "name": "To-do",
    "taskIds": Array [],
  },
  "11": Object {
    "id": 11,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "12": Object {
    "id": 12,
    "isCompletedColumn": true,
    "name": "Done",
    "taskIds": Array [],
  },
  "2": Object {
    "id": 2,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [
      5,
    ],
  },
  "3": Object {
    "id": 3,
    "name": "To-do",
    "taskIds": Array [
      6,
    ],
  },
  "4": Object {
    "id": 4,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "5": Object {
    "id": 5,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
  "6": Object {
    "id": 6,
    "name": "To-do",
    "taskIds": Array [],
  },
  "7": Object {
    "id": 7,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "8": Object {
    "id": 8,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
}
`;

exports[`columns reducer DELETE_COLUMN 1`] = `
Object {
  "0": Object {
    "id": 0,
    "name": "To-do",
    "taskIds": Array [
      0,
      1,
      2,
    ],
  },
  "1": Object {
    "id": 1,
    "name": "In Progress",
    "taskIds": Array [
      3,
      4,
    ],
  },
  "10": Object {
    "id": 10,
    "name": "To-do",
    "taskIds": Array [],
  },
  "11": Object {
    "id": 11,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "12": Object {
    "id": 12,
    "isCompletedColumn": true,
    "name": "Done",
    "taskIds": Array [],
  },
  "2": Object {
    "id": 2,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [
      5,
    ],
  },
  "3": Object {
    "id": 3,
    "name": "To-do",
    "taskIds": Array [
      6,
    ],
  },
  "4": Object {
    "id": 4,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "5": Object {
    "id": 5,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
  "6": Object {
    "id": 6,
    "name": "To-do",
    "taskIds": Array [],
  },
  "7": Object {
    "id": 7,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "8": Object {
    "id": 8,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
}
`;

exports[`columns reducer MOVE_TASK_VERTICALLY 1`] = `
Object {
  "0": Object {
    "id": 0,
    "name": "To-do",
    "taskIds": Array [
      0,
      1,
    ],
  },
  "1": Object {
    "id": 1,
    "name": "In Progress",
    "taskIds": Array [
      3,
      4,
      7,
    ],
  },
  "10": Object {
    "id": 10,
    "name": "To-do",
    "taskIds": Array [],
  },
  "11": Object {
    "id": 11,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "12": Object {
    "id": 12,
    "isCompletedColumn": true,
    "name": "Done",
    "taskIds": Array [],
  },
  "2": Object {
    "id": 2,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [
      5,
      1,
    ],
  },
  "3": Object {
    "id": 3,
    "name": "To-do",
    "taskIds": Array [
      6,
    ],
  },
  "4": Object {
    "id": 4,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "5": Object {
    "id": 5,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
  "6": Object {
    "id": 6,
    "name": "To-do",
    "taskIds": Array [],
  },
  "7": Object {
    "id": 7,
    "name": "In Progress",
    "taskIds": Array [],
  },
  "8": Object {
    "id": 8,
    "isCompletedColumn": true,
    "name": "Complete",
    "taskIds": Array [],
  },
}
`;
